### PR TITLE
Availability: Fixes region failover logic on control plane hot path when gateway hangs

### PR DESCRIPTION
**Pull request by ealsur**
If Gateway is a hung state where the HttpRequestException only happens after 60 seconds, the current Hot Path policy was only waiting up to 10 seconds and surfacing a RequestTimeout, and we could not apply the failover mechanics.

This only affected the hot path (address resolution or query plans) and not other Gateway calls.

Introduced in 3.16.0 in PR https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1954
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/HttpClient/HttpTimeoutPolicyControlPlaneRetriableHotPath.cs [hunk 1]
```diff
@@ -22,6 +22,7 @@ private readonly IReadOnlyList<(TimeSpan requestTimeout, TimeSpan delayForNextRe
             (TimeSpan.FromSeconds(.5), TimeSpan.Zero),
             (TimeSpan.FromSeconds(5), TimeSpan.FromSeconds(1)),
             (TimeSpan.FromSeconds(10), TimeSpan.Zero),
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/HttpClient/HttpTimeoutPolicyControlPlaneRetriableHotPath.cs:24]: Should the 10 second just be converted to 65 seconds? I'm just thinking if it didn't resolve in 5 seconds I doubt the 10 seconds will work.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/HttpClient/HttpTimeoutPolicyControlPlaneRetriableHotPath.cs:24]: Makes sense, done

**Review by j82w — APPROVED** [review]

**Review by kirankumarkolli — CHANGES_REQUESTED** [review]
Lets please wait merging this change.

Latency is also equally very important as well. 
May be we need to revisit basing all fail-over on just ReqeustException

**Comment by j82w** [comment]
> Lets please wait merging this change.
> 
> Latency is also equally very important as well.
> May be we need to revisit basing all fail-over on just ReqeustException

@kirankumarkolli  I don't see any reason to block this PR. If it is decided that we want to revisit fail-over design it should be a separate PR after the current model is fixed.

**Comment by ealsur** [comment]
I don't see how latency is involved here. We are already retrying 3 times, with .5, and 5 seconds, before the 65 seconds. If Gateway has a latency spike, it will be clear.
The 3 retries are to work around Gateway upgrades and try to reach other instances, not for latency (rather availability).

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**